codegenerator/cli/src/config_parsing/system_config.rs

```diff
@@ -267,49 +278,218 @@ impl SystemConfig {
+                            ))?;
+                        }
+                    }
+                }
+            }
 
-        let schema =
-            Schema::parse_from_file(&schema_path).context("Parsing schema file for config")?;
+            let sync_source = SyncSource::HyperfuelConfig(HyperfuelConfig {
+                endpoint_url: "https://fuel-testnet.hypersync.xyz".to_string(),
+            });
 
-        Self::parse_from_human_cfg_with_schema(human_cfg, schema, project_paths)
+            let contracts: Vec<NetworkContract> = network
+                .contracts
+                .iter()
+                .cloned()
+                .map(|c| NetworkContract {
+                    name: c.name,
+                    addresses: c.address.into(),
+                })
+                .collect();
+
+            let network = Network {
+                id: network.id as u64,
+                start_block: network.start_block,
+                end_block: network.end_block,
+                confirmed_block_threshold: DEFAULT_CONFIRMED_BLOCK_THRESHOLD,
+                sync_source,
+                contracts,
+            };
+
+            unique_hashmap::try_insert(&mut networks, network.id.clone(), network)
+                .context("Failed inserting network at networks map")?;
+        }
+
+        Ok(SystemConfig {
+            name: fuel_config.name.clone(),
+            ecosystem: Ecosystem::Fuel,
+            parsed_project_paths: project_paths.clone(),
+            schema_path: fuel_config
+                .schema
+                .clone()
+                .unwrap_or_else(|| DEFAULT_SCHEMA_PATH.to_string()),
+            networks,
+            contracts,
+            unordered_multichain_mode: false,
+            rollback_on_reorg: false,
+            save_full_history: false,
+            schema,
+            field_selection: FieldSelection::empty(),
+            enable_raw_events: false,
+        })
+    }
+
+    pub fn parse_from_project_files(project_paths: &ParsedProjectPaths) -> Result<Self> {
+        let human_config_string =
+            std::fs::read_to_string(&project_paths.config).context(format!(
+          "EE104: Failed to resolve config path {0}. Make sure you're in the correct directory and \
+           that a config file with the name {0} exists",
+          &project_paths.config
+              .to_str()
+              .unwrap_or("{unknown}"),
+        ))?;
+
+        let config_discriminant: human_config::ConfigDiscriminant =
+          serde_yaml::from_str(&human_config_string)
+                .context("EE105: Failed to deserialize config. The config.yaml file is either not a valid yaml or the \"ecosystem\" field is not a string.")?;
+
+        let ecosystem = match config_discriminant.ecosystem {
+            Some(ecosystem) if ecosystem == "evm" => Ecosystem::Evm,
+            Some(ecosystem) if ecosystem == "fuel" => Ecosystem::Fuel,
```

Comment:
A nice trick for matching on strings

```suggestion
        let ecosystem = match config_discriminant.ecosystem.as_ref() {
            Some("evm") => Ecosystem::Evm,
            Some("fuel") => Ecosystem::Fuel,
```

Reply:
Nice

Reply:
@JonoPrest It wanted `as_deref`